File: oneInchAPI.py
import requests
import os
from dotenv import load_dotenv

# Load environment variables from .env file
load_dotenv()

API = os.getenv("API_TOKEN")
PSM = "0x17A8541B82BF67e10B0874284b4Ae66858cb1fd5"
USDC = "0xaf88d065e77c8cC2239327C5EDb3A432268e5831"
WETH = "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1"
ETH = "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"
USDT = "0xFd086bC7CD5C481DCC9C85ebE478A1C0b69FCbb9"
USDCE = "0xFF970A61A04b1cA14834A43f5dE4533eBDDB5CC8"

PSM_AMOUNT = 59410765679358251996998  # 59410765679358251996998

method = "get"
apiUrl = "https://api.1inch.dev/swap/v5.2/42161/swap"

requestOptions = {
    "headers": {"Authorization": f"Bearer {API}"},
    "body": {},
    "params": {
        "src": f"{PSM}",
        "dst": f"{ETH}",
        "amount": f"{PSM_AMOUNT}",
        "from": "0x5615dEB798BB3E4dFa0139dFa1b3D433Cc23b72f",
        "slippage": "10",
        "receiver": "0x3440326f551B8A7ee198cEE35cb5D517f2d296a2",
        "disableEstimate": "true",
        "compatibility": "true",
        "protocols": "ARBITRUM_UNISWAP_V3",
    },
}

# Prepare request components
headers = requestOptions.get("headers", {})
body = requestOptions.get("body", {})
params = requestOptions.get("params", {})

response = requests.get(apiUrl, headers=headers, params=params)

# Check if the request was successful (status code 200)
if response.status_code == 200:
    # Parse the JSON response
    json_response = response.json()

    # Access the 'data' field
    data_field = json_response.get("tx").get("data")[10:]

    print(data_field)
else:
    print(f"Request failed with status code {response.json()}")


# if it will be used for converting portal energy to liquidity. minPSM, and minWeth should be added to the end of calldata.

File: oneInchAPI.js
// curl version 
// curl -X GET \
//       "https://api.1inch.dev/swap/v6.0/42161/swap?src=0x17A8541B82BF67e10B0874284b4Ae66858cb1fd5&compatibility=true&disableEstimate=true&allowPartialFill=false&usePermit2=false&slippage=5&from=0xD59Eb7E224Ad741C06c26d4670Fc0C2D89121DE3&dst=0x82aF49447D8a07e3bd95BD0d56f35241523fBab1&amount=25000000000000000000000&receiver=0xB9927a561527Ac7Bb7a93cDc80ba3c7F14EBDD1e&" \
//       -H "Authorization: Bearer oESfoRrx7Qu0xEWUf3M9WCs7oRRYsrap" \
//       -H "accept: application/json" \
//       -H "content-type: application/json" \


// you just need to change 
// slippage
// from: contract address
// dst: to token
// amount (for adding liquidity call just divide amount by 2)
// receiver


// for node js type
// first, install node
// then, npm install axios
// then, npm install dotenv (see .envExample for creating .env file)
// then, node file.js


const dotenv = require('dotenv');
dotenv.config();
const axios = require("axios");

async function httpCall() {

    const url = "https://api.1inch.dev/swap/v6.0/42161/swap";

    const config = {
        headers: {
            "Authorization": `Bearer ${process.env.API_TOKEN}`
        },
        params: {
            "src": "0x17A8541B82BF67e10B0874284b4Ae66858cb1fd5",
            "dst": "0x82aF49447D8a07e3bd95BD0d56f35241523fBab1",
            "amount": "25000000000000000000000",
            "from": "0x35f0DB0b238D3d0653d75FB67Cd64eE65C32DeFc",
            "slippage": "1",
            "receiver": "0x35f0DB0b238D3d0653d75FB67Cd64eE65C32DeFc",
            "allowPartialFill": "false",
            "disableEstimate": "true",
            "usePermit2": "false",
            "compatibility": "true"
        }
    };


    try {
        const response = await axios.get(url, config);
        // Extract tx from response
        const txData = response.data.tx.data;
        const cutData = "0x" + txData.substring(10);
        console.log(cutData);
    } catch (error) {
        console.error(error);
    }
}

httpCall()




